ethcore/sync/src/snapshot_sync.rs

```diff
@@ -22,20 +22,43 @@ use keccak_hash::keccak;
 use log::trace;
 use snapshot::SnapshotService;
 use common_types::snapshot::ManifestData;
+use indexmap::IndexSet;
 
 #[derive(PartialEq, Eq, Debug)]
+/// The type of data contained in a chunk: state or block.
 pub enum ChunkType {
+	/// The chunk contains state data (aka account data).
 	State(H256),
+	/// The chunk contains block data.
 	Block(H256),
 }
 
 #[derive(Default, MallocSizeOf)]
 pub struct Snapshot {
-	pending_state_chunks: Vec<H256>,
-	pending_block_chunks: Vec<H256>,
+	/// List of hashes of the state chunks we need to complete the warp sync from this snapshot.
+	/// These hashes are contained in the Manifest we downloaded from the peer(s).
+	/// Note: this is an ordered set so that state restoration happens in order, which keeps
+	/// memory usage down.
+	// See https://github.com/paritytech/parity-common/issues/255
+	#[ignore_malloc_size_of = "no impl for IndexSet (yet)"]
+	pending_state_chunks: IndexSet<H256>,
+	/// List of hashes of the block chunks we need to complete the warp sync from this snapshot.
+	/// These hashes are contained in the Manifest we downloaded from the peer(s).
+	/// Note: this is an ordered set so that state restoration happens in order, which keeps
+	/// memory usage down.
+	// See https://github.com/paritytech/parity-common/issues/255
+	#[ignore_malloc_size_of = "no impl for IndexSet (yet)"]
+	pending_block_chunks: IndexSet<H256>,
```

Comment:
I'm not sure it's worth the extra dependency on `indexmap`. The only benefit is removing the chunk for the set when it's done.